build-system/tasks/3p-vendor-helpers.js

```diff
@@ -76,34 +61,54 @@ async function buildVendorConfigs(options) {
 }
 
 /**
- * Build the JavaScript for the vendor specified
- *
- * @param {string} name Name of the extension. Must be the sub directory in
- *     the extensions directory and the name of the JS and optional CSS file.
- * @param {!Object} options
- * @return {!Promise}
+ * Generate bundles for all JS to be built.
+ * @return {Object}
  */
-async function buildVendor(name, options) {
-  await compileJs(
-    './3p/vendors/',
-    name + '.js',
-    './dist.3p/',
-    Object.assign(options, {
-      include3pDirectories: true,
-      includePolyfills: true,
-      externs: ['./ads/ads.extern.js'],
-      toName: `current/vendor/${name}.max.js`,
-      minifiedName: `${VERSION}/vendor/${name}.js`,
-      esmPassCompilation: false,
-    })
-  );
+function generateBundles() {
+  const bundles = {};
+  listVendors().forEach((vendor) => {
+    bundles[`${vendor}`] = {
+      srcDir: './3p/vendors/',
+      srcFilename: `${vendor}.js`,
+      destDir: './dist.3p/current/vendor/',
+      minifiedDestDir: `./dist.3p/${VERSION}/vendor/`,
+      options: {
+        include3pDirectories: true,
+        includePolyfills: true,
+        externs: ['./ads/ads.extern.js'],
+        toName: `${vendor}.max.js`,
+        minifiedName: `${vendor}.js`,
+        esmPassCompilation: false,
+      },
+    };
+  });
+  return bundles;
+}
```

Comment:
Looks like this function iterates through all 3p vendor bundles for every call of `lazyBuild3pVendor()`. Can its output be cached so it's generated only once?

Reply:
Updated: Cached the vendor bundle.